dbms/src/Common/UniqueLockGuard.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <mutex>
+
+template <typename Mutex>
+struct UniqueLockGuard
```

Comment:
Seems not necessary? Since there is a `get_lock` function to wrap the logic of acquiring the lock. `auto lock = get_lock()` is simple enough.

Reply:
Currently I use get_lock to try to reduce the overhead in single thread scene

Reply:
> Seems not necessary? Since there is a `get_lock` function to wrap the logic of acquiring the lock. `auto lock = get_lock()` is already simple enough.

I will remove the UniqueLockGuard

Reply:
> Currently I use get_lock to try to reduce the overhead in single thread scene

Got it. LGTM.